qml/keys/CharKey.qml

```diff
@@ -101,12 +116,16 @@ Item {
 
     Text {
         id: keyLabel
-        text: (panel.activeKeypadState === "NORMAL") ? label : shifted;
-        anchors.centerIn: parent
+        text: (imgNormal === UI.imageGreyKey[formFactor]) ? label : (panel.activeKeypadState === "NORMAL") ? label : shifted;
+        anchors.horizontalCenter: buttonImage.horizontalCenter
+		anchors.verticalCenter: buttonImage.verticalCenter 
+		anchors.verticalCenterOffset: imgNormal === UI.imageGreyKey[formFactor] ? units.gu(0.5) : units.gu(-0.25)
         font.family: UI.fontFamily
-        font.pixelSize: fontSize
+        font.pixelSize: ((imgNormal === UI.imageGreyKey[formFactor])&&(panel.activeKeypadState !== "NORMAL") ) ? UI.annotationFontSize : fontSize
```

Comment:
This takes care of the "special keys" like , and . since they behave differently. I didn't want to create a new key kind having 98% the same code ;)